File: algorithms/1_easy/M/03200_maximum_height_of_a_triangle/solution.py
class Solution:
    def maxHeightOfTriangle(self, red: int, blue: int) -> int:
        curr_height = 0
        odd_count = 0
        even_count = 0
        curr_level = 1

        while True:
            if (curr_level & 1) == 1:
                odd_count += curr_level
            else:
                even_count += curr_level

            if not ((odd_count <= red and even_count <= blue) or (odd_count <= blue and even_count <= red)):
                return curr_level - 1

            curr_level += 1

File: algorithms/1_easy/M/03200_maximum_height_of_a_triangle/solution.rs
impl Solution {
    pub fn max_height_of_triangle(red: i32, blue: i32) -> i32 {
        let mut curr_height = 0_i32;
        let mut odd_count = 0_i32;
        let mut even_count = 0_i32;
        let mut curr_level = 1_i32;

        loop {
            if (curr_level & 1) == 1 {
                odd_count += curr_level;
            } else {
                even_count += curr_level;
            }

            if !((odd_count <= red && even_count <= blue) || (odd_count <= blue && even_count <= red)) {
                return curr_level - 1;
            }

            curr_level += 1;
        }
    }
}
